OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -192,6 +195,12 @@ void UpdateCondition(Actor self)
 
 		protected virtual void Tick(Actor self)
 		{
+			if (reportLoop != null && reportTick < self.World.WorldTick)
```

Comment:
Isn't cancelling the sound potentially in the middle of the track going to be really harsh?
IMO a better solution, depending on the source track, would be to play the track without looping and then check `reportLoop.Complete` here and play it again until `reportTick < self.World.WorldTick`.  This will guarantee that it will always play through a complete number of times, with a tradeoff of introducing a delay of up to 40ms between repeats.

Reply:
This doesn't seem to be resolved.